Index: linux-5.4.213/arch/arm64/boot/dts/allwinner/Makefile
===================================================================
--- linux-5.4.213.orig/arch/arm64/boot/dts/allwinner/Makefile
+++ linux-5.4.213/arch/arm64/boot/dts/allwinner/Makefile
@@ -18,6 +18,8 @@ dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-li
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-nanopi-neo2.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-nanopi-neo-plus2.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-pc2.dtb
+dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-pc2-spinand.dtb
+dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-pc2-driver.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-prime.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-zero-plus.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-orangepi-zero-plus2.dtb
