Index: u-boot-2020.07/arch/arm/dts/Makefile
===================================================================
--- u-boot-2020.07.orig/arch/arm/dts/Makefile
+++ u-boot-2020.07/arch/arm/dts/Makefile
@@ -570,6 +570,7 @@ dtb-$(CONFIG_MACH_SUN8I_R40) += \
 	sun8i-v40-bananapi-m2-berry.dtb
 dtb-$(CONFIG_MACH_SUN8I_V3S) += \
 	sun8i-v3s-licheepi-zero.dtb \
+	sun8i-v3s-licheepi-zero-spinand.dtb \
 	sun8i-s3-pinecube.dtb \
 	sun8i-v3s-sinlinx-sinv3s.dtb
 dtb-$(CONFIG_MACH_SUN50I_H5) += \
