Index: u-boot-2022.01/tools/Makefile
===================================================================
--- u-boot-2022.01.orig/tools/Makefile
+++ u-boot-2022.01/tools/Makefile
@@ -172,9 +172,9 @@ endif
 
 # MXSImage needs LibSSL
 ifneq ($(CONFIG_MX23)$(CONFIG_MX28)$(CONFIG_ARMADA_38X)$(CONFIG_TOOLS_LIBCRYPTO),)
-HOSTCFLAGS_kwbimage.o += \
+HOSTCFLAGS_kwbimage.o += -lpthread \
 	$(shell pkg-config --cflags libssl libcrypto 2> /dev/null || echo "")
-HOSTLDLIBS_mkimage += \
+HOSTLDLIBS_mkimage += -lpthread \
 	$(shell pkg-config --libs libssl libcrypto 2> /dev/null || echo "-lssl -lcrypto")
 
 # OS X deprecate openssl in favour of CommonCrypto, supress deprecation
